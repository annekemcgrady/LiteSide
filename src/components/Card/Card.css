.card {
  height: 280px;
  width: 200px;
  /* border: 1px solid red; */
  margin: 10px;
}
.card-header {
  background-color: #000;
  border: 1px solid #3e3e3e;
  color: #ffff2b;
  font-size: 1.1em;
  height: 20%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.card-body {
  background: rgba(0,0,0,0.19);
  border: 1px solid #3e3e3e;
  color: white;
  height: 65%;
  text-align: left;
}

.card-footer {
  background-color: rgba(0,0,0,0.19);
  border: 1px solid #3e3e3e;
  height: 15%;
}

.card-fav-btn {
  width: 30px;
  float: right;
  cursor: pointer;
  padding-right: 7px;
  padding-top: 8px;
}

.body-item {
  padding: 5px;
  padding-left: 12px;
}

.label {
  color: rgb(165, 165, 165);
  display: block;
  font-size: 12px;
}

