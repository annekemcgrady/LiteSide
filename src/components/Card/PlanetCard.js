import React from 'react';

const PlanetCard = ({ planet }) => {
  return (
    <section className="card">
      <h2 className="card-header">{planet.name}</h2>
      <div className="card-body">
        <p>Terrain: {planet.terrain}</p>
        <p>Diameter: {planet.diameter}</p>
        <p>Population: {planet.population}</p>
      </div>
      <footer className="card-footer"></footer>
    </section>
  )
}

export default PlanetCard;