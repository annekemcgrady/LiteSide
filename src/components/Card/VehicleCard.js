import React from 'react';


const VehicleCard = ({ vehicle }) => {
  return (
    <section className="card">
      <h2 className="card-header">{person.name}</h2>
      <div className="card-body">
        <p>Model: {vehicle.model}</p>
        <p>Class: {vehicle.vehicle_class}</p>
        <p>Number of Passengers: {vehicle.passengers}</p>
      </div>
      <footer className="card-footer"></footer>
    </section>
  )
}

export default VehicleCard;