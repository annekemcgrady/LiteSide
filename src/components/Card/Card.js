import React from 'react';
import './Card.css';
import star from '../../images/star.png';
import starActive from '../../images/star-active.png';


class Card extends React.Component {

  Items = () => {
    const {item} = this.props;
    if (item.type === 'people') {

      return (
        <>
          <p className="body-item"><span className="label">Birth Year</span>{item.birth_year}</p>
          <p className="body-item"><span className="label">Gender</span>{item.gender}</p>
          <p className="body-item"><span className="label">Height</span>{item.height}</p>
          <p className="body-item"><span className="label">Eye Color</span>{item.eye_color}</p>
        </>
      )
    }
    if (item.type === 'planets') {

      return (
        <>
          <p className="body-item"><span className="label">Terrain</span>{item.terrain}</p>
          <p className="body-item"><span className="label">Diameter</span>{item.diameter}</p>
          <p className="body-item"><span className="label">Population</span>{item.population}</p>
        </>
      )
    }
    if (item.type === 'vehicles') {

      return (
        <>
          <p className="body-item"><span className="label">Model</span>{item.model}</p>
          <p className="body-item"><span className="label">Class</span>{item.vehicle_class}</p>
          <p className="body-item"><span className="label">Number of Passengers</span>{item.passengers}</p>
        </>
      )
    }
  }

   changeFav = (e) => {
    this.props.handleFavorite(e.target.id)
  }
  render () {
    const {favorite} = this.props.item
    const starType = !favorite ? star : starActive;
    return (
      <section className="card">
        <h3 className="card-header">{this.props.item.name}</h3>
        <div className="card-body">
          {this.Items()}
        </div>
        <footer className="card-footer">
          <img className="car-fav-btn" src={starType} onClick={this.changeFav} id={this.props.item.id} alt="favorite button"/>

        </footer>
      </section>
    )
  }
  
}



export default Card;