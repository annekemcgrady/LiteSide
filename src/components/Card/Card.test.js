import React from 'react';
import { shallow, mount } from 'enzyme';
import Card from './Card';


describe('Card', ()=> {
let fakeData;

beforeEach(()=> {
  fakeData = {
    birth_year: "19BBY",
    created: "2014-12-09T13:50:51.644000Z",
    edited: "2014-12-20T21:17:56.891000Z",
    eye_color: "blue",
    favorite: true,
    films: (5)["https://swapi.co/api/films/2/", "https://swapi.co/api/films/6/", "https://swapi.co/api/films/3/", "https://swapi.co/api/films/1/", "https://swapi.co/api/films/7/"],
    gender: "male",
    hair_color: "blond",
    height: "172",
    homeworld: "https://swapi.co/api/planets/1/",
    id: "pp-0",
    mass: "77",
    name: "Luke Skywalker",
    skin_color: "fair",
    species: ["https://swapi.co/api/species/1/"],
    starships: (2) ["https://swapi.co/api/starships/12/", "https://swapi.co/api/starships/22/"],
    type: "people",
    url: "https://swapi.co/api/people/1/",
    vehicles: (2) ["https://swapi.co/api/vehicles/14/", "https://swapi.co/api/vehicles/30/"]
  }
});

  it ('should match the snapshot', () => {
    const wrapper = shallow(<Card key={fakeData.id} item={fakeData} handleFavorite={jest.fn()} isFavorited={false} />)
    expect(wrapper).toMatchSnapshot;
  })

  it('should match the favorited snapshot', () => {
    const wrapper = shallow(<Card key={fakeData.id} item={fakeData} handleFavorite={jest.fn()} isFavorited={true} />)
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should run handleFavorite prop when the Favorite button is clicked', () => {
    const wrapper = shallow(<Card key={fakeData.id} item={fakeData} handleFavorite={jest.fn()} isFavorited={true} />)
    wrapper.instance().handleFavorite = jest.fn();
    wrapper.find('.card-fav-btn').simulate('click');
    expect(wrapper.instance().handleFavorite).toHaveBeenCalledWith('pp-00');
  });

});
