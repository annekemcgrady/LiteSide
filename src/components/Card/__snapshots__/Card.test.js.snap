// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the favorited snapshot 1`] = `
<section
  className="card"
>
  <h3
    className="card-header"
  >
    Luke Skywalker
  </h3>
  <div
    className="card-body"
  >
    <p
      className="body-item"
    >
      <span
        className="label"
      >
        Birth Year
      </span>
      19BBY
    </p>
    <p
      className="body-item"
    >
      <span
        className="label"
      >
        Gender
      </span>
      male
    </p>
    <p
      className="body-item"
    >
      <span
        className="label"
      >
        Height
      </span>
      172
    </p>
    <p
      className="body-item"
    >
      <span
        className="label"
      >
        Eye Color
      </span>
      blue
    </p>
  </div>
  <footer
    className="card-footer"
  >
    <img
      alt="favorite button"
      className="card-fav-btn"
      id="pp-0"
      onClick={[Function]}
      src="star-active.png"
    />
  </footer>
</section>
`;

exports[`Card should match the snapshot 1`] = `ShallowWrapper {}`;
